subsys/bluetooth/audio/ascs.c

```diff
@@ -1001,19 +1001,14 @@ static void ase_disable(struct bt_ascs_ase *ase)
 static void disconnected(struct bt_conn *conn, uint8_t reason)
 {
 	struct bt_ascs *session = &sessions[bt_conn_index(conn)];
+	struct bt_ascs_ase *ase, *tmp;
 
 	if (session->conn == NULL) {
 		return;
 	}
 
-	sys_snode_t *ase_node, *s;
-
-	SYS_SLIST_FOR_EACH_NODE_SAFE(&session->ases, ase_node, s) {
-		struct bt_bap_stream *stream;
-		struct bt_ascs_ase *ase;
-
-		ase = CONTAINER_OF(ase_node, struct bt_ascs_ase, node);
-		stream = ase->ep.stream;
+	SYS_SLIST_FOR_EACH_CONTAINER_SAFE(&session->ases, ase, tmp, node) {
+		struct bt_bap_stream *stream = ase->ep.stream;
```

Comment:
Am I missing something here? 
The _SAFE macros make sense to use when removing something from the list, but from what I can tell, the ASES are not being removed from `session` here, so can't we just just the non-SAFE version?

Reply:
Ah, yes, because `ase_release -> ascs_ep_set_state(BT_BAP_EP_STATE_RELEASING) -> ascs_ep_set_state(ep, BT_BAP_EP_STATE_IDLE) -> ase_free` 

Given the fairly long list of function calls here (including a recursive one) where the change to `BT_BAP_EP_STATE_IDLE` is conditional, I think we should add a comment here :)